tests/foreman/ui_airgun/test_computeresource.py

```diff
@@ -125,219 +151,160 @@ def test_positive_delete(session):
-    assert not session.computeresource.vm_status(rhev['Name'], vm_name)
+        session.computeresource.create({
+            'name': name,
+            'provider': FOREMAN_PROVIDERS['rhev'],
+            'provider_content.url': rhev_data['rhev_url'],
+            'provider_content.user': rhev_data['username'],
+            'provider_content.password': rhev_data['password'],
+            'provider_content.api4': version,
+            'provider_content.datacenter.value': rhev_data['datacenter'],
+            'provider_content.certification_authorities': module_ca_cert
+        })
+        vm = session.computeresource.list_vms(name, rhev_data['vm_name'])
+        assert vm['Name'].read() == rhev_data['vm_name']
 
 
 @tier2
-def test_positive_v3_wui_virtual_machines_can_be_powered_on(
-        session, module_ca_cert):
-    """Power on a machine on RHEV using APIv3
-
-    :id: f75e994a-6da1-40a3-9685-42387388b308
-    """
-    with session:
-        rhev3 = add_rhev(session, 3, module_ca_cert)
-        vm_poweron(session, rhev3)
+def test_positive_edit_resource_version(session, module_ca_cert, rhev_data):
+    """Edit RHEV Compute Resource with another protocol version
 
+    :id: 6e7985b6-a605-4fb8-8710-17a046bdac53
 
-@tier2
-def test_positive_v4_wui_virtual_machines_can_be_powered_on(
-        session, module_ca_cert):
-    """Power on a machine on RHEV using APIv4
+    :expectedresults: resource updated successfully and switched to another
+        protocol version
 
-    :id: f75e994a-6da1-40a3-9685-42387388b309
+    :CaseLevel: Integration
     """
+    name = gen_string('alpha')
     with session:
-        rhev4 = add_rhev(session, 4, module_ca_cert)
-        vm_poweron(session, rhev4)
+        session.computeresource.create({
+            'name': name,
+            'provider': FOREMAN_PROVIDERS['rhev'],
+            'provider_content.url': rhev_data['rhev_url'],
+            'provider_content.user': rhev_data['username'],
+            'provider_content.password': rhev_data['password'],
+            'provider_content.api4': False,
+            'provider_content.datacenter.value': rhev_data['datacenter'],
+            'provider_content.certification_authorities': module_ca_cert
+        })
+        resource_values = session.computeresource.read(name)
+        assert not resource_values['provider_content']['api4']
+        session.computeresource.edit(name, {'provider_content.api4': True})
+        resource_values = session.computeresource.read(name)
+        assert resource_values['provider_content']['api4']
 
 
 @tier2
-def test_positive_v3_wui_virtual_machines_can_be_powered_off(
-        session, module_ca_cert):
-    """Power off a machine on RHEV using APIv3
-
-    :id: f75e994a-6da1-40a3-9685-42387388b30a
-    """
-    with session:
-        rhev3 = add_rhev(session, 3, module_ca_cert)
-        vm_poweroff(session, rhev3)
+@parametrize('version', [True, False])
+def test_positive_resource_vm_power_management(
```

Comment:
perhaps this test has to be run in one thread.
 are we sure it will run in same process? 
as one test can change the state, when the other is running.